Connection timeouts to Sonatype

I'm not sure this is a bug as such, but figured I'd raise this.

I've been seeing a lot of connection timeouts lately while uploading to Sonatype, especially from less-powerful machines (i.e. from GitHub Actions).

I _think_ the issue is that this plugin opens up a connection to Sonatype, and _then_ runs Dokka/assemble, waiting for all of the artifacts to upload. Dokka 1.4.0+ is pretty slow, so we can easily timeout the connection (which I assume is ~30 seconds). This is just from my observations though, it might not actually be that.

I have a workaround in https://github.com/chrisbanes/accompanist/pull/146, which basically runs Dokka + assemble _before_ `uploadArchives`: `./gradlew assembleRelease dokkaHtml uploadArchives`. 

Maybe the task dependency graph for `uploadArchives` could be tweaked to achieve similar?

Here's a example log:

```
> Task :insets:preBuild UP-TO-DATE
> Task :insets:preDebugBuild UP-TO-DATE
> Task :insets:compileDebugRenderscript NO-SOURCE
> Task :insets:generateDebugResValues FROM-CACHE
> Task :insets:generateDebugResources UP-TO-DATE
> Task :insets:packageDebugResources FROM-CACHE
> Task :insets:parseDebugLocalResources FROM-CACHE
> Task :insets:processDebugManifest FROM-CACHE
> Task :insets:generateDebugRFile FROM-CACHE
> Task :insets:preReleaseBuild UP-TO-DATE
> Task :insets:compileReleaseRenderscript NO-SOURCE
> Task :insets:generateReleaseResValues FROM-CACHE
> Task :insets:generateReleaseResources UP-TO-DATE
> Task :insets:packageReleaseResources FROM-CACHE
> Task :insets:parseReleaseLocalResources FROM-CACHE
> Task :insets:processReleaseManifest FROM-CACHE
> Task :insets:generateReleaseRFile FROM-CACHE

> Task :insets:dokkaHtml
Dokka 1.4.* is an alpha project
Initializing plugins
Creating documentation models
Transforming documentation model before merging
Merging documentation models
Transforming documentation model after merging
Creating pages
Transforming pages
Rendering


> Task :insets:dokkaHtml FAILED


FAILURE: Build failed with an exception.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.

* What went wrong:
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7/userguide/command_line_interface.html#sec:command_line_warnings
129 actionable tasks: 32 executed, 97 from cache
Execution failed for task ':insets:dokkaHtml'.
> java.net.SocketTimeoutException: Read timed out
```

--- a/nexus/src/main/kotlin/com/vanniktech/maven/publish/nexus/Nexus.kt
+++ b/nexus/src/main/kotlin/com/vanniktech/maven/publish/nexus/Nexus.kt
@@ -1,6 +1,7 @@
 package com.vanniktech.maven.publish.nexus
 
 import java.io.IOException
+import java.util.concurrent.TimeUnit
 import okhttp3.OkHttpClient
 import retrofit2.Retrofit
 import retrofit2.converter.moshi.MoshiConverterFactory
@@ -14,6 +15,9 @@ internal class Nexus(
   private val service by lazy {
     val okHttpClient = OkHttpClient.Builder()
       .addInterceptor(NexusOkHttpInterceptor(username, password))
+      .connectTimeout(30, TimeUnit.SECONDS)
+      .readTimeout(30, TimeUnit.SECONDS)
+      .writeTimeout(30, TimeUnit.SECONDS)
       .build()
     val retrofit = Retrofit.Builder()
       .addConverterFactory(MoshiConverterFactory.create())

--- a/nexus/src/main/kotlin/com/vanniktech/maven/publish/nexus/Nexus.kt
+++ b/nexus/src/main/kotlin/com/vanniktech/maven/publish/nexus/Nexus.kt
@@ -16,9 +16,9 @@ internal class Nexus(
   private val service by lazy {
     val okHttpClient = OkHttpClient.Builder()
       .addInterceptor(NexusOkHttpInterceptor(username, password))
-      .connectTimeout(30, TimeUnit.SECONDS)
-      .readTimeout(30, TimeUnit.SECONDS)
-      .writeTimeout(30, TimeUnit.SECONDS)
+      .connectTimeout(60, TimeUnit.SECONDS)
+      .readTimeout(60, TimeUnit.SECONDS)
+      .writeTimeout(60, TimeUnit.SECONDS)
       .build()
     val retrofit = Retrofit.Builder()
       .addConverterFactory(MoshiConverterFactory.create())

--- a/nexus/src/main/kotlin/com/vanniktech/maven/publish/nexus/Nexus.kt
+++ b/nexus/src/main/kotlin/com/vanniktech/maven/publish/nexus/Nexus.kt
@@ -2,6 +2,7 @@ package com.vanniktech.maven.publish.nexus
 
 import java.io.IOException
 import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
 import okhttp3.OkHttpClient
 import retrofit2.Retrofit
 import retrofit2.converter.moshi.MoshiConverterFactory
@@ -118,6 +119,8 @@ internal class Nexus(
           break
         }
       } catch (e: IOException) {
+        System.err.println("Exception trying to get repository status: ${e.message}")
+      } catch (e: TimeoutException) {
         System.err.println("Exception trying to get repository status: ${e.message}")
       }
     }

